src/anyvar/storage/snowflake.py

```diff
@@ -0,0 +1,479 @@
+
+            _logger.debug(
+                "Connecting to Snowflake account %s with params %s",
+                account_name,
+                sanitized_conn_params,
+            )
+        # log connection attempt
+        else:
+            _logger.info(
+                "Connecting to Snowflake account %s",
+                account_name,
+            )
+
+        # create the database connection and ensure it is setup
+        self.conn = snowflake.connector.connect(account=account_name, **conn_params)
+        self.ensure_schema_exists()
+
+        # setup batch handling
+        self.batch_manager = SnowflakeBatchManager
+        self.batch_mode = False
+        self.batch_insert_values = []
+        self.batch_limit = batch_limit or int(
+            os.environ.get("ANYVAR_SNOWFLAKE_STORE_BATCH_LIMIT", "100000")
+        )
+        max_pending_batches = max_pending_batches or int(
+            os.environ.get("ANYVAR_SNOWFLAKE_STORE_MAX_PENDING_BATCHES", "50")
+        )
+        self.batch_thread = SnowflakeBatchThread(self.conn, self.table_name, max_pending_batches)
+        self.batch_thread.start()
+
+    def _create_schema(self):
+        """Add the VRS object table if it does not exist"""
+        # self.table_name is only modifiable via environment variable or direct instantiation of the SnowflakeObjectStore
+        create_statement = f"""
+        CREATE TABLE {self.table_name} (
+            vrs_id VARCHAR(500) PRIMARY KEY,
+            vrs_object VARIANT
+        );
+        """  # nosec B608
+        _logger.info("Creating VRS object table %s", self.table_name)
+        with self.conn.cursor() as cur:
+            cur.execute(create_statement)
+
+    def ensure_schema_exists(self):
+        """Check that VRS object table exists and create it if it does not"""
+        with self.conn.cursor() as cur:
+            cur.execute(
+                f"""
+                SELECT COUNT(*) FROM information_schema.tables 
+                 WHERE table_catalog = CURRENT_DATABASE() AND table_schema = CURRENT_SCHEMA() 
+                 AND UPPER(table_name) = UPPER('{self.table_name}');
+                """  # nosec B608
+            )
+            result = cur.fetchone()
+
+        if result is None or result[0] <= 0:
+            self._create_schema()
+
+    def __repr__(self):
+        return str(self.conn)
+
+    def __setitem__(self, name: str, value: Any):
+        """Add item to database. If batch mode is on, add item to batch and submit batch
+        for write only if batch size is exceeded.
+
+        :param name: value for `vrs_id` field
+        :param value: value for `vrs_object` field
+        """
+        assert ga4gh.core.is_pydantic_instance(value), "ga4gh.vrs object value required"
+        name = str(name)  # in case str-like
+        if self.batch_mode:
+            self.batch_insert_values.append((name, value))
+            if _logger.isEnabledFor(logging.DEBUG):
```

Comment:
small nitpick: this `isEnabled` call, as well as the one 16 lines down, seem unnecessary, since the only computation is in the logging step itself (and `_logger.debug()` shouldn't perform string interpolation unless debug logging is enabled)